sdcm/cluster.py

```diff
@@ -1736,18 +1742,15 @@ def install_manager_agent(self, package_path=None):
                 self.parent_cluster.params.get("scylla_mgmt_agent_repo",
                                                self.parent_cluster.params.get("scylla_mgmt_repo", None)))
             package_name = 'scylla-manager-agent'
-        package_mgr = "yum" if self.distro.is_rhel_like else "apt-get"
         install_and_config_agent_command = dedent(r"""
             {} install -y {}
-            sed -i 's/# auth_token:.*$/auth_token: {}/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
             scyllamgr_ssl_cert_gen
-            sed -i 's/#tls_cert_file/tls_cert_file/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
-            sed -i 's/#tls_key_file/tls_key_file/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
-            sed -i 's/#prometheus: .*/prometheus: :{}/' /etc/scylla-manager-agent/scylla-manager-agent.yaml
-            systemctl restart scylla-manager-agent
-            systemctl enable scylla-manager-agent
-        """.format(package_mgr, package_name, auth_token, manager_prometheus_port))
+        """.format("yum" if self.distro.is_rhel_like else "apt-get", package_name))
         self.remoter.run('sudo bash -cxe "%s"' % install_and_config_agent_command)
+        with self.remote_manager_yaml(path=package_name) as manager_agent_yaml:
+            manager_agent_yaml["auth_token"] = manager_agent_yaml["tls_cert_file"] = \
```

Comment:
this totally incorrect, read the sed command again, it's removing things from comments.
not exactly what you are doing here, assigning `/etc/scylla-manager-agent/scylla-manager-agent.yaml` to all parameters....

Reply:
Yes, you are right, I did not understand the meaning of the `sed` command.
Last night I looked at the command and changed it...
Despite this, the run is failed.